### ISSUE TITLE
regression: needless_continue triggers within a macro

### ISSUE THREAD

**Issue by Indeximal** [rust-lang/rust-clippy#16175]
### Summary

I am using `darling` which has macros that apparently trigger the `clippy::needless_continue` lint. However, this is hidden in the current version clippy 0.1.91 (ed61e7d7e2 2025-11-07), but is reported in clippy 0.1.92 (f3f12444a0 2025-11-09). If I read [the docs](https://doc.rust-lang.org/nightly/clippy/development/macro_expansions.html#false-positives) this should be a false positive. Or was this was intentionally changed?

### Lint Name

needless_continue

### Reproducer

I tried this code:

```rust
#[derive(darling::FromVariant)]
#[darling(attributes(can))]
pub struct CanDecodeVariant {
    ident: syn::Ident,
}
```

which expands to:
<details>

```rust
#[automatically_derived]
#[allow(clippy::manual_unwrap_or_default)]
impl ::darling::FromVariant for Foo {
    fn from_variant(__variant: &::darling::export::syn::Variant) -> ::darling::Result<Self> {
        let mut __errors = ::darling::Error::accumulator();
        use ::darling::ToTokens;
        for __attr in &__variant.attrs {
            match ::darling::export::ToString::to_string(&__attr.path().clone().into_token_stream())
                .as_str()
            {
                "foo" => match ::darling::util::parse_attribute_to_meta_list(__attr) {
                    ::darling::export::Ok(__data) => {
                        match ::darling::export::NestedMeta::parse_meta_list(__data.tokens) {
                            ::darling::export::Ok(ref __items) => {
                                if __items.is_empty() {
                                    continue;
                                }
                                for __item in __items {
                                    match *__item {
                                        ::darling::export::NestedMeta::Meta(ref __inner) => {
                                            let __name =
                                                ::darling::util::path_to_string(__inner.path());
                                            match __name.as_str() {
                                                __other => {
                                                    __errors.push(
                                                        ::darling::Error::unknown_field(__other)
                                                            .with_span(__inner),
                                                    );
                                                }
                                            }
                                        }
                                        ::darling::export::NestedMeta::Lit(ref __inner) => {
                                            __errors.push(
                                                ::darling::Error::unsupported_format("literal")
                                                    .with_span(__inner),
                                            );
                                        }
                                    }
                                }
                            }
                            ::darling::export::Err(__err) => {
                                __errors.push(__err.into());
                            }
                        }
                    }
                    ::darling::export::Err(__err) => {
                        __errors.push(__err);
                    }
                },
                _ => continue,
            }
        }
        __errors.finish()?;
        ::darling::export::Ok(Self {
            ident: __variant.ident.clone(),
        })
    }
}
```

</details>

I saw this happen when running `cargo +beta clippy`:

```
warning: this `continue` expression is redundant
  --> datatypes/src/lib.rs:82:10
   |
82 | #[derive(darling::FromVariant)]
   |          ^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider dropping the `continue` expression
   = help: for further information visit https://rust-lang.github.io/rust-clippy/beta/index.html#needless_continue
   = note: `-W clippy::needless-continue` implied by `-W clippy::pedantic`
   = help: to override `-W clippy::pedantic` add `#[allow(clippy::needless_continue)]`
   = note: this warning originates in the derive macro `darling::FromVariant` (in Nightly builds, run with -Z macro-backtrace for more info)
```

When I run `cargo clippy`, it doesn't report this error though.

### Version

```text
ustc 1.92.0-beta.3 (f3f12444a 2025-11-09)
binary: rustc
commit-hash: f3f12444a017add0468f683f3a10656d29438a05
commit-date: 2025-11-09
host: x86_64-apple-darwin
release: 1.92.0-beta.3
LLVM version: 21.1.3
```

**Comment by relaxcn** [comment]
I think maybe we can't deal with this problem from our side.

The `dealing` proc-macro uses  `def_site` , so that's why the lint can't detect if the code comes from macro expansion.

**Comment by Indeximal** [comment]
Hm oke, why does it change between beta and stable then? But yeah, then you can close this issue, I just allowlisted the lint

**Comment by m4tx** [comment]
I can confirm this affects me as well with the clippy version currently in stable (`0.1.92 (ded5c06cf2 2025-12-08)`).
